lib/rubygems/sigstore/rekor/api.rb

```diff
@@ -42,13 +42,14 @@ def where(data_digest:)
       raise "Unexpected response from POST /api/v1/index/retrieve:\n #{retrieve_response}"
     end
 
-    retrieve_response.body.map do |uuid|
-      entry_response = connection.get("api/v1/log/entries/#{uuid}")
-      unless entry_response.status == 200
-        raise "Unexpected response from GET api/v1/log/entries/#{uuid}:\n #{entry_response}"
-      end
+    retrieve_response = connection.post("api/v1/log/entries/retrieve", entryUUIDs: retrieve_response.body)
```

Comment:
We're doing this pattern alot:
```ruby
connection.verb(some_url, body)
unless response.status is expected
  raise "Unexpected response from VERB some_url:\n #{response}"
end
```

Some of this could be extracted into utility code. I'll create a low-priority issue.